src/pages/index.js

```diff
@@ -50,51 +82,132 @@ const editModal = new ModalWithForm(
 
 editModal.setEventListeners();
 
+function handleProfileFormSubmit(data) {
+  editModal.setLoading(true);
+  api
+    .setUserInfo(data)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editModal.setLoading(false);
+    });
+  editModal.close();
+}
+
+// Add Card Modal
 const newCardModal = new ModalWithForm(
   {
     modalSelector: "#card__add-modal",
   },
   handleCardFormSubmit
 );
-newCardModal.setEventListeners();
 
-const userInfo = new UserInfo({
-  nameSelector: ".profile__title",
-  descriptionSelector: ".profile__description",
-});
+newCardModal.setEventListeners();
 
-function fillProfileForm() {
-  const info = userInfo.getUserInfo();
-  profileTitleInput.value = info.name;
-  profileDescriptionInput.value = info.description;
+function handleCardFormSubmit(data) {
+  newCardModal.setLoading(true);
+  api
+    .addCard(data)
+    .then((res) => {
+      const card = new Card(
+        res,
+        "#card__template",
+        handleImageClick,
+        handleDeleteClick,
+        handleLikeClick
+      );
+      cardList.prepend(card.getCard());
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      newCardModal.setLoading(false);
+    });
+  newCardModal.close();
 }
 
-// Handlers
-function handleProfileFormSubmit(data) {
-  userInfo.setUserInfo(data);
-  editModal.close();
+function handleLikeClick(card) {
+  if (card.isLiked) {
+    api
+      .removeLike(card.id)
+      .then((res) => {
+        card.handleLikeIcon(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .likeCard(card.id)
+      .then((res) => {
+        card.handleLikeIcon(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors